packages/task/src/browser/task-service.ts

```diff
@@ -997,24 +993,19 @@ export class TaskService implements TaskConfigurationClient {
         terminal.sendText(selectedText);
     }
 
-    async attach(terminalId: number, taskId: number): Promise<void> {
-        // Get the list of all available running tasks.
-        const runningTasks: TaskInfo[] = await this.getRunningTasks();
-        // Get the corresponding task information based on task id if available.
-        const taskInfo: TaskInfo | undefined = runningTasks.find((t: TaskInfo) => t.taskId === taskId);
+    async attach(taskInfo: AttachableTaskInfo): Promise<void> {
```

Comment:
This is a breaking change.
If it is necessary it should be included in the `CHANGELOG`.